graphene_django_extras/mutation.py

```diff
@@ -1,306 +1,315 @@
+        data = kwargs.get(cls._meta.input_field_name)
+        request_type = info.context.META.get("CONTENT_TYPE", "")
+        if "multipart/form-data" in request_type:
+            data.update({name: value for name, value in info.context.FILES.items()})
+
+        nested_objs = cls.manage_nested_fields(data, root, info)
+        serializer = cls._meta.serializer_class(
+            data=data, **cls.get_serializer_kwargs(root, info, **kwargs)
+        )
+
+        ok, obj = cls.save(serializer, root, info)
+        if not ok:
+            return cls.get_errors(obj)
+        elif nested_objs:
+            [getattr(obj, field).add(*objs) for field, objs in nested_objs.items()]
+        return cls.perform_mutate(obj, info)
+
+    @classmethod
+    def delete(cls, root, info, **kwargs):
+        pk = kwargs.get("id")
+
+        old_obj = get_Object_or_None(cls._meta.model, pk=pk)
+        if old_obj:
+            old_obj.delete()
+            old_obj.id = pk
+            return cls.perform_mutate(old_obj, info)
+        else:
+            return cls.get_errors(
+                [
+                    ErrorType(
+                        field="id",
+                        messages=[
+                            "A {} obj with id {} do not exist".format(
+                                cls._meta.model.__name__, pk
+                            )
+                        ],
+                    )
+                ]
+            )
+
+    @classmethod
+    def update(cls, root, info, **kwargs):
+        data = kwargs.get(cls._meta.input_field_name)
+        request_type = info.context.META.get("CONTENT_TYPE", "")
+        if "multipart/form-data" in request_type:
+            data.update({name: value for name, value in info.context.FILES.items()})
+
+        pk = data.pop("id")
+        old_obj = get_Object_or_None(cls._meta.model, pk=pk)
+        if old_obj:
+            nested_objs = cls.manage_nested_fields(data, root, info)
+            serializer = cls._meta.serializer_class(
+                old_obj,
+                data=data,
+                partial=True,
+                **cls.get_serializer_kwargs(root, info, **kwargs),
+            )
+
+            ok, obj = cls.save(serializer, root, info)
+            if not ok:
+                return cls.get_errors(obj)
+            elif nested_objs:
+                [getattr(obj, field).add(*objs) for field, objs in nested_objs.items()]
+            return cls.perform_mutate(obj, info)
+        else:
+            return cls.get_errors(
+                [
+                    ErrorType(
+                        field="id",
+                        messages=[
+                            "A {} obj with id: {} do not exist".format(
+                                cls._meta.model.__name__, pk
+                            )
+                        ],
+                    )
+                ]
+            )
+
+    @classmethod
+    def save(cls, serialized_obj, root, info, **kwargs):
+        """
```

Comment:
I honestly don't know why the whole file was rewritten. I tried both in vim and vscode, and in both the diff showed a full rewrite.

Interesting part below, explained in the comment. 
Maybe there is a better place to check for Enum in values @eamigo86 ?